src/miner.cpp

```diff
@@ -624,6 +623,104 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
+    int64_t nActualStakeOutputValue = 0;
+
+    // Pull out total value of output for stake, which also includes (interest or CBR) and research rewards at this point.
+    nTotalStakeOutputValue = blocknew.vtx[1].vout[1].nValue;
+    nUTXOCount = GetNumberofStakeOutputs(nTotalStakeOutputValue);
+    
+    // Only go to the effort to split if nUTXOCount > 1, otherwise leave undisturbed.
+    if (nUTXOCount > 1)
+    {
+        scriptPubKey = blocknew.vtx[1].vout[1].scriptPubKey;
+        
+        // Remove the existing single stake output.
+        blocknew.vtx[1].vout.pop_back();
+    
+        nActualStakeOutputValue = nTotalStakeOutputValue / nUTXOCount;
+
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nUTXOCount = %f", nUTXOCount);
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nActualStakeOutputValue = %f", CoinToDouble(nActualStakeOutputValue));
+        
+        int64_t nSumStakeUTXOValue = 0;
+        for (unsigned int i = 1; i < nUTXOCount; i++)
+        {
+            blocknew.vtx[1].vout.push_back(CTxOut(nActualStakeOutputValue, scriptPubKey));
+            LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", i, CoinToDouble(nActualStakeOutputValue));
+            nSumStakeUTXOValue += nActualStakeOutputValue;
+        }
+        // For the last UTXO, subtract the running sum from the desired total, which does not include the last UTXO so far,
+        // to recover anything lost from rounding. This will be a very small difference from the others.
+        // The reason we go through this trouble is that integer division rounds down, and we don't even want
+        // to have the wallet lose 1 Halford.
+        blocknew.vtx[1].vout.push_back(CTxOut((nTotalStakeOutputValue - nSumStakeUTXOValue), scriptPubKey));
+        LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", nUTXOCount, CoinToDouble(nTotalStakeOutputValue - nSumStakeUTXOValue));
+    }
+}
+
+
+
+unsigned int GetNumberofStakeOutputs(int64_t nValue)
+{
+    int64_t nMinUTXOPostSplitValue = 0;
+    double dEfficiency = 0.0;
+    int64_t nDesiredStakeUTXOValue = 0;
+    unsigned int nUTXOCount = 1;
+
+    // Pull Minimum Post Stake UTXO Split Value from config or command line parameter.
+    // Do not allow it to be specified below 800 GRC.
+    nMinUTXOPostSplitValue = GetArg("-minimumstakeutxovalueforsplit", 800) * COIN;
+    nMinUTXOPostSplitValue = (nMinUTXOPostSplitValue < 800 * COIN) ? 800 * COIN : nMinUTXOPostSplitValue;
+
+    LogPrintf("GetNumberofStakeOutputs: nMinUTXOPostSplitValue = %f", CoinToDouble(nMinUTXOPostSplitValue));
+
+    // Pull efficiency for UTXO staking from config, but constrain to the interval [0.75, 0.98]. Use default of 0.90.
+    dEfficiency = (double)GetArg("-stakingefficiency", 90) / 100;
+    if (dEfficiency > 0.98)
+        dEfficiency = 0.98;
+    else if (dEfficiency < 0.75)
+        dEfficiency = 0.75;
+
+    LogPrintf("GetNumberofStakeOutputs: dEfficiency = %f", dEfficiency);
```

Comment:
Please remove prints like this, or at least condition under fDebug2/fDebug

Reply:
Then you should remove the original prints from CreateCoinStake, which are similar.

Reply:
I put them under fDebug2.

Reply:
If we are going to be consistent with the LogPrintf's in CreateCoinStake (at 605 for example), then the ones at 663 and 671 should remain non-conditional. I have conditioned all others on fDebug2.